EndlessClient/Controllers/LoginController.cs

```diff
@@ -127,47 +129,47 @@ public async Task LoginToCharacter(ICharacter character)
                 if (unableToLoadMap || _fileRequestActions.NeedsFileForLogin(InitFileType.Map, _currentMapStateProvider.CurrentMapID))
                 {
                     gameLoadingDialog.SetState(GameLoadingDialogState.Map);
-                    if (!await SafeGetFile(async () => await _fileRequestActions.GetMapFromServer(_currentMapStateProvider.CurrentMapID)))
+                    if (!await SafeGetFile(async () => await _fileRequestActions.GetMapFromServer(_currentMapStateProvider.CurrentMapID, sessionID)))
```

Comment:
For these lines, do this instead of creating an async lambda:
```csharp
if (!await SafeGetFile(() => _fileRequestActions.GetMapFromServer(_currentMapStateProvider.CurrentMapID, sessionID)))
    return;
```
It reduces the overhead in the async state machine when you return the task directly, since it is eventually await anyway

Reply:
Done :)